sim/deathknight/dps/rotation_frost_helper.go

```diff
@@ -71,111 +75,112 @@ func (dk *DpsDeathknight) RotationActionCallback_EndOfFightPrio(sim *core.Simula
 	obAt := core.MaxDuration(frAt, uhAt)
 	fsCost := float64(core.RuneCost(dk.FrostStrike.CurCast.Cost).RunicPower())
 	delayAmount := core.MinDuration(time.Duration(dk.Rotation.OblitDelayDuration)*time.Millisecond, 2501*time.Millisecond)
-	bothblAt := dk.BloodRuneBothReadyAt()
+	bothblAt := dk.BloodDeathRuneBothReadyAt()
+	spellHitcap := true
 
-	if dk.Talents.Epidemic == 2 {
+	if bothblAt > 9223372036854775800*time.Nanosecond {
```

Comment:
<img width="588" alt="Screen Shot 2022-12-07 at 11 26 32 am" src="https://user-images.githubusercontent.com/114305673/206056653-4bf0dbb4-5604-40fc-9760-e24866890e58.png">

if both blood runes are death runes, then if 1 of them is ready (292 years never expires) return the min duration, the other rune (so 1 is ready the time to wait is the 2nd rune), if both are ready it'll return never expires. the 1st else returns the greater duration if both are on CD, else return -1 is for when they aren't death runes.

knowing this, when it calls the function
both death, 1 regenning: returns the regenning rune
both death, both ready: returns never expires
both death, both regenning: returns the later regenned rune
at least 1 isn't death, returns -1

because the code uses timestamp comparison, if it returns never expires it will never use this time for obAt, so it won't cast obliterate with double death runes. that's why there's the 

if bothblAt > 9223372036854775800*time.Nanosecond {
	bothblAt = 1
}

so it will always return true when comparing against sim.CurrentTime

if bothblAt == -1 {
	bothblAt = 9223372036854775800 * time.Nanosecond
}

is for when they aren't death, DON'T TRY TO QUEUE OBLIT

if dk.Talents.Epidemic == 2 || diseaseExpiresAt >= simDur {
	obAt = core.MinDuration(obAt, bothblAt)
}
this is for either unh sub to use death rune obliterates always, or for blood sub to use death rune obliterate when the diseases last past the end of the fight (no need to spend blood runes for pestilence anymore)

Reply:
ok that was really dumb, I did a new commit with proper constants

Reply:
we also have core.NeverExpires as a time that always in the future

Reply:
<img width="647" alt="Screen Shot 2022-12-09 at 10 30 42 am" src="https://user-images.githubusercontent.com/114305673/206588651-933f120b-829c-4907-9248-5a8d3215ad42.png">
yeah that's what the last commit did, hooper talked some sense into me about that